### ISSUE TITLE
Entity inventory

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#6081]
I suggest that Lua entities can have an inventory, similar to what players and nodes already have.

---
Note: I posted this to keep this *feature request* separate from the *documentation bug report* https://github.com/minetest/minetest/issues/6079.

**Comment by C1ffisme** [comment]
Detached inventories in Minetest are about as fun to work with as molten lava is with bare hands. Any work towards making NPC or other entity inventories as easy to make as node inventories is heroic to me.

**Comment by paramat** [comment]
1 year bump for core dev opinions.

Does this mean detached inventories can supply this feature?

**Comment by Wuzzy2** [comment]
Giving objects an inventory just makes infinitely more sense than having to fiddle around with detached inventories and keeping track of to which object they belong to.
I believe you could somehow hack together a detached-inventory-for-object hackery, but this just feels wrong and is quite ugly.

It's a glaring omission IMO. Players have inventories, but normal objects don't. Mobs definitely could use inventories (think of a donkey with chests).

**Comment by sfan5** [comment]
The existence of workaround does not save you from implementing something, this is a valid and justified feature request.

**Comment by paramat** [comment]
Yeah, seems a potentially good idea, i was just unsure how easy this was to do with detached inventories.
sfan5, note my neutral question above, i wasn't suggesting the workaround should be used instead, i was asking, and i didn't even label this possible close, but was bumping for attention, you are over-reacting.

**Comment by vladtcvs** [comment]
I'm trying to implement this feature.

**Comment by runsy** [comment]
I would like this feature too. For a mob improvement.

**Comment by The4codeblocks** [comment]
by now we can just name an entityinv `detached:entity_<GUID>`
though being able to have an optional attached inventory to the entity would be nice
